docs/csharp/roslyn-sdk/tutorials/how-to-write-csharp-analyzer-code-fix.md

```diff
@@ -0,0 +1,505 @@
+---
+title: "Tutorial: Write your first analyzer and code fix"
+description: This tutorial provides step-by-step instructions to build an analyzer and code fix using the .NET Compiler SDK (Roslyn APIs).
+ms.date: 08/01/2018
+ms.custom: mvc
+---
+# Tutorial: Write your first analyzer and code fix
+
+The .NET Compiler Platform SDK provides the tools you need to create custom warnings that target C# or Visual Basic code. Your **analyzer** contains code that recognizes violations of your rule. Your **code fix** contains the code that fixes the violation. The rules you implement can be anything from code structure to coding style to naming conventions and more. The .NET Compiler Platform provides the framework for running analysis as developers are writing code, and all the Visual Studio UI features for fixing code: showing squiggles in the editor, populating the Visual Studio Error List, creating the "light bulb" suggestions and showing the rich preview of the suggested fixes.
+
+In this tutorial, you'll explore the creation of an **analyzer** and an accompanying **code fix** using the Roslyn APIs. An analyzer is a way to perform source code analysis and report a problem to the user. Optionally, an analyzer can also provide a code fix that represents a modification to the user's source code. This tutorial creates an analyzer that finds local variable declarations that could be declared using the `const` modifier but are not. The accompanying code fix modifies those declarations to add the `const` modifier.
+
+## Prerequisites
+
+* [Visual Studio 2017](https://www.visualstudio.com/downloads)
+
+You'll need to install the **.NET Compiler Platform SDK**:
+
+[!INCLUDE[interactive-note](~/includes/roslyn-installation.md)]
+
+There are several steps to creating and validating your analyzer:
+
+1. Create the solution.
+1. Register the analyzer name and description.
+1. Report analyzer warnings and recommendations.
+1. Implement the code fix to accept recommendations.
+1. Improve the analysis through unit tests.
+
+## Explore the analyzer template
+
+Your analyzer reports to the user any local variable declarations that can be converted to local constants. For example, consider the following code:
```

Comment:
I was just thinking about this part of the section. It talks about how my analyzer reports any local vars that can be converted to consts. This section is named explore the template, the template doesn't do this (yet) and the next thing mentioned is running VS and seeing how it tells the user the class has lowercase letters. I think these intro paragraphs and samples should be moved down to the **Create analyzer registrations**  section